tests/core/test_processor.py

```diff
@@ -207,7 +205,20 @@
     assert len(t.events) == 3  # nothing should have been executed
 
 
-async def test_reminder_cancelled(
+async def wait_for_all_jobs(timeout_after_seconds: Optional[float] = None) -> None:
+    total_seconds = 0.0
+    while len((await jobs.scheduler()).get_jobs()) > 0 and (
+        total_seconds < timeout_after_seconds or not timeout_after_seconds
```

Comment:
I think `not timeout_after_seconds` has to be the first part of the condition because `total_seconds < None` probably throws an error, right?